model/DataBaseAccess.php

```diff
@@ -177,7 +178,28 @@ public function addMultiplePermissions(array $permissionData)
                 }
             }
         }
-        $this->getPersistence()->insertMultiple(self::TABLE_PRIVILEGES_NAME, $insert);
+
+        $this->getLogger()->debug(
+            "Processing {count} permission inserts in {chunks} chunks",
+            [
+                'count' => count($insert),
+                'chunks' => ceil(count($insert)/self::INSERT_CHUNK_SIZE)
+            ]
+        );
+
+        foreach(array_chunk($insert, self::INSERT_CHUNK_SIZE) as $index => $batch) {
+            $this->getLogger()->debug(
```

Comment:
It's not so critical, but maybe we can store `getLogger` in a separate variable or cache this call?

```php
    public function getLogger()
    {
        if ($this->logger instanceof LoggerInterface) {
            return $this->logger;
        }
        if ($this instanceof ServiceLocatorAwareInterface) {
            $logger = $this->getServiceLocator()->get(LoggerService::SERVICE_ID);
        } else {
            $logger = ServiceManager::getServiceManager()->get(LoggerService::SERVICE_ID);
        }
        return ($logger instanceof LoggerInterface) ? $logger : new NullLogger();
    }
```

As you can see, if logger is an instance of `LoggerInterface` - we are good. But in other cases, we will use `ServiceLocator`/`ServiceManager` to get the necessary service. And this might impact a bit on performance.

But as I said, this is not so critical, and you can skip this one 🙂😅

Reply:
That's a good point, changed in b95ca785c1c655a49c2e1fa8725e365b78b54a99 to use a single reference to the logger through the method. Thanks :slightly_smiling_face: